.github/workflows/builds.yaml

```diff
@@ -0,0 +1,105 @@
+name: Build Tests
+
+on:
+  schedule:
+    - cron: "0 2 * * *"
+  push:
+    branches: [main]
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test_gcc:
+    runs-on: [self-hosted]
+    matrix:
+      gcc: [9.5.0, 10.5.0, 12.3.0, 13.2.0]
```

Comment:
Do this instead:
```
compiler:
  - gcc/9.5.0
  - gcc/10.5.0
  - gcc/12.3.0
  - gcc/13.2.0
  - clang/12.0.1
  - clang/14.0.6
  - clang/16.0.6
  - clang/17.0.6
  - intel/oneAPI-2021.4
```

Reply:
I think I correctly addressed all of the comments. Let's me know what you think of the latest version. I also removed the use of `configure.sh`.